geode-core/src/main/java/org/apache/geode/internal/cache/LocalRegion.java

```diff
@@ -1439,83 +1437,113 @@ Object nonTxnFindObject(KeyInfo keyInfo, boolean isCreate, boolean generateCallb
       Object localValue, boolean disableCopyOnRead, boolean preferCD,
       ClientProxyMembershipID requestingClient, EntryEventImpl clientEvent,
       boolean returnTombstones) throws TimeoutException, CacheLoaderException {
-
     @Retained
+    Object result;
+    if (isProxy()) {
```

Comment:
isProxy just covers region with the empty data policy.
What about PR accessors (i.e. localMaxMemory==0)? Do they end up using this code or do they use a different code path?

Reply:
They use a different code path.